tinygrad/codegen/uops.py

```diff
@@ -224,12 +224,11 @@ def get_recursive_parents(x:UOp, with_phi=False) -> Set[UOp]:
     replaced_stores: Dict[UOp,UOp] = {}
     for u in self.uops:
       if u.uop is not UOps.STORE or (val:=u.vin[-1]).uop is not UOps.CAST or cast(DType,val.dtype).count == 1: continue
-      if u.vin[0].uop is UOps.DEFINE_LOCAL: continue # TODO add support for local store
       if all(el.uop is UOps.GEP for el in val.vin): replaced_stores[u] = val.vin[0].vin[0]
       elif all(el.uop is UOps.PHI for el in val.vin): replaced_stores[u] = phi_resolve_acc(val)
     for prev,new in replaced_stores.items():
-      self.add(UOps.STORE, prev.dtype, (prev.vin[0],prev.vin[1],new), insert_before=self.uops.index(prev))
-      self.uops.remove(prev)
+      self.uops.remove(prev.vin[-1]) # remove the old upcast
```

Comment:
Is this a childless uop? You might be able to do this optimization before the remove childless pass, then it will be autoremoved.

Reply:
that won't work because of the PHI support.
If there's a `cast(acc.x, acc.y, acc.z, acc.w)` it's casting 4 PHI nodes. It's their only parent so if I let the childless pass run it'll remove those PHIs too.